x/logic/keeper/grpc_query_ask_test.go

```diff
@@ -137,16 +139,18 @@ func TestGRPCAsk(t *testing.T) {
 				maxGas:        1000,
 				expectedError: "out of gas: logic <ReadPerByte> (1018/1000): limit exceeded",
 			},
+			{
+				query:         "bank_balances(X, Y).",
+				maxGas:        3000,
+				expectedError: "error executing query: out of gas in location: panic: {ValuePerByte}: out of gas: invalid argument",
+			},
 			{
 				query:  "block_height(X).",
 				maxGas: 3000,
 				predicateCosts: map[string]uint64{
 					"block_height/1": 10000,
 				},
-				expectedAnswer: &types.Answer{
-					Variables: []string{"X"},
-					Results:   []types.Result{{Error: "error(resource_error(gas(block_height/1,12353,3000)),block_height/1)"}},
-				},
+				expectedError: "error executing query: out of gas in location: block_height/1: out of gas: invalid argument",
```

Comment:
I think wherever the out of gas occur the returned error should be a `limit exceeded` like the first test case, this one is particularly helpful when investigating failed transaction as it appears clearly in the logs